ui/app/components/secret-edit-toolbar.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * @module SecretEditToolbar
+ * SecretEditToolbar component is the toolbar component displaying the JSON toggle and the actions like delete in the show mode.
+ *
+ * @example
+ * ```js
+ * <SecretEditToolbar
+ * @mode={{mode}}
+ * @model={{this.model}}
+ * @isV2={{isV2}}
+ * @isWriteWithoutRead={{isWriteWithoutRead}}
+ * @secretDataIsAdvanced={{secretDataIsAdvanced}}
+ * @showAdvancedMode={{showAdvancedMode}}
+ * @modelForData={{this.modelForData}}
+ * @navToNearestAncestor={{this.navToNearestAncestor}}
+ * @canUpdateSecretData={{canUpdateSecretData}}
+ * @codemirrorString={{codemirrorString}}
+ * @wrappedData={{wrappedData}}
+ * @editActions={{hash
+    toggleAdvanced=(action "toggleAdvanced")
+    refresh=(action "refresh")
+  }}
+ * />
+ * ```
+ 
+ * @param {string} mode - show, create, edit. The view.
+ * @param {object} model - the model passed from the parent secret-edit
+ * @param {boolean} isV2 - KV type
+ * @param {boolean} isWriteWithoutRead - boolean describing permissions
+ * @param {boolean} secretDataIsAdvanced - used to determine if show JSON toggle
+ * @param {boolean} showAdvacnedMode - used for JSON toggle
+ * @param {object} modelForData - a modified version of the model with secret data
+ * @param {string} navToNearestAncestor - route to nav to if press cancel
+ * @param {boolean} canUpdateSecretData - permissions that show the create new version button or not.
+ * @param {string} codemirrorString - used to copy the JSON
+ * @param {object} wrappedData - when copy the data it's the token of the secret returned.
+ * @param {object} editActions - actions passed from parent to child
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { not } from '@ember/object/computed';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+export default class SecretEditToolbar extends Component {
```

Comment:
Again, all these functions came from secretEdit. I just broke out the ones relevant to the toolbar.